### PR TITLE
feat: Implementation of query and variable editor for GraphQL datasource

### LINKED ISSUE #10550: [Feature]: Create the GraphQL Query Editor Page with Query and Variable Inputs
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

- [ ] Create a GraphQL Query Editor Page
- [ ] Add Query and Variable Inputs for the GraphQL requests

### Why should this be worked on?

This is the important part of creating a GraphQL queries for the user.

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

This PR introduces : 
- a further flow after the creation of datasource for Graphql
- different type of Codemirror editors for query and variable values
- Implements the evaluated values for query and variable keys
- Resizeable variable and query editors
- Coloring of Graphql query syntax
- Default configuration when a Graphql request is created

Fixes #10550 , #10551

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Manually

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/pages/Editor/APIEditor/GraphQL/VariableEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  display: flex;
+  width: 100%;
+  background: ${Colors.GREY_2};
+  padding: 8px 16px;
+`;
+
+const EXPECTED_VARIABLE = {
+  type: "object",
+  example:
+    '{\n  "name":"{{ inputName.property }}",\n  "preference":"{{ dropdownName.property }}"\n}',
```
Comment (rashmi-rai) [inline app/client/src/pages/Editor/APIEditor/GraphQL/VariableEditor.tsx:35]: example: `{{ “name”: inputName.property, “preference”: dropdownName.property }}` . 
for prettified evaluated value.

File: app/client/src/pages/Editor/APIEditor/GraphQL/VariableEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        </TooltipComponent>
+      </VariableHeader>
+      <DynamicTextField
+        border={CodeEditorBorder.ALL_SIDE}
+        dataTreePath={`${props.actionName}.config.variable`}
+        expected={EXPECTED_VARIABLE}
+        height="100%"
+        mode={EditorModes.JSON_WITH_BINDING}
+        name="actionConfiguration.variable"
+        placeholder={`${EXPECTED_VARIABLE.example}\n\n\\\\Take widget inputs using {{ }}`}
```
Comment (rashmi-rai) [inline app/client/src/pages/Editor/APIEditor/GraphQL/VariableEditor.tsx:72]: `placeholder={EXPECTED_VARIABLE.example}`

File: app/client/src/pages/Editor/APIEditor/GraphQL/VariableEditor.tsx (hunk 1 above)
Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/APIEditor/GraphQL/VariableEditor.tsx:35]: It's just an example to state that the dynamic bindings can be used inside variable Editor by using `{{`

Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/APIEditor/GraphQL/VariableEditor.tsx:35]: Wrapping whole of the object to `{{` would evaluate the whole expression whether or not it has the dynamic values. So I think wrapping individual values to `{{` would be a better approach. Open to suggestions. Would change the lines if anything else is decided after the performance thing is measured. Till then resolving the conversation.

**Review by rashmi-rai — APPROVED** [review]

**Pull request merged**